File: smart-contracts/calc-contract2.py
#!/usr/bin/python
from __future__ import print_function
from datetime import date
from money import Money
from dateutil.relativedelta import relativedelta
from decimal import Decimal
import findates
from ContractRevenue import LoanContract
import LoanCalculator

calculator = LoanCalculator.LoanCalculator()
calculator.test_wrapper()
exit

print("Standard Payments")
my_contract = LoanContract()
my_contract.set_events({
    "revenues":[],
    "early_payments":[],
    "credit_draws":[]})
calculator = LoanCalculator.LoanCalculator()
calculator.calculate(my_contract)
print()

print("Assume revenue hit on 2015-05-15")
my_contract = LoanContract()
my_contract.set_events({
    "revenues": [
    {"on" : date(2015, 5,15), "amount" : Money("800000", "HKD")}
    ],
    "early_payments":[],
    "credit_draws":[]})
calculator = LoanCalculator.LoanCalculator()
calculator.calculate(my_contract)
print()
print("Assume revenue hit on 2015-06-15 and 2015-09-15")
my_contract = LoanContract()
my_contract.set_events({
    "revenues": [
    {"on" : date(2015, 6,15), "amount" : Money("800000", "HKD")},
    {"on" : date(2015, 9,15), "amount" : Money("800000", "HKD")}
    ],
    "early_payments":[],
    "credit_draws":[]})
calculator = LoanCalculator.LoanCalculator()
calculator.calculate(my_contract)
print()
print("Assume revenue hit on 2015-06-15 and 2015-09-15 and credit draws")
my_contract = LoanContract()
my_contract.set_events({
    "revenues": [
    {"on" : date(2015, 6,15), "amount" : Money("800000", "HKD")},
    {"on" : date(2015, 9,15), "amount" : Money("800000", "HKD")}
    ],
    "early_payments":[
    {"on": date(2015,4,15), "amount": Money(5000, "HKD")}
    ],
    "credit_draws":[
    {"on": date(2015,8,15), "amount": Money(15000, "HKD")}
    ]})
calculator = LoanCalculator.LoanCalculator()
calculator.calculate(my_contract)
print()

File: smart-contracts/calc-contract2.js
#!/usr/bin/node
var LoanCalculator = require("./LoanCalculator.js");
var TermSheet = require("./TermSheet.js");
var moment = require("moment");
var Decimal = require("decimal");

function money(a, b) {
    return {"amount": new Decimal(a), "ccy" : b};
}

calculator = new LoanCalculator.LoanCalculator();
calculator.test_wrapper()

console.log("Standard Payments")
my_term_sheet = new TermSheet.TermSheet();
my_term_sheet.set_events({
    "revenues":[],
    "early_payments":[],
    "credit_draws":[]})
calculator = new LoanCalculator.LoanCalculator()
calculator.calculate(my_term_sheet)
console.log()

console.log("Assume revenue hit on 2015-05-15")
my_term_sheet =  new TermSheet.TermSheet();
my_term_sheet.set_events({
    "revenues": [
	{"on" : moment("2015-05-15"),
	 "amount" : money("800000", "HKD")}
    ],
    "early_payments":[],
    "credit_draws":[]})
calculator = new LoanCalculator.LoanCalculator()
calculator.calculate(my_term_sheet)
console.log()

console.log("Assume revenue hit on 2015-06-15 and 2015-09-15")
my_term_sheet = new TermSheet.TermSheet()
my_term_sheet.set_events({
    "revenues": [
    {"on" : moment("2015-06-15"),
     "amount" : money("800000", "HKD")},
    {"on" : moment("2015-09-15"), 
     "amount" : money("800000", "HKD")}
    ],
    "early_payments":[],
    "credit_draws":[]})
calculator = new LoanCalculator.LoanCalculator()
calculator.calculate(my_term_sheet)
console.log()
console.log("Assume revenue hit on 2015-06-15 and 2015-09-15 and credit draws")
my_term_sheet = new TermSheet.TermSheet();
my_term_sheet.set_events({
    "revenues": [
    {"on" : moment("2015-06-15"),
     "amount" : money("800000", "HKD")},
    {"on" : moment("2015-09-15"),
     "amount" : money("800000", "HKD")}
    ],
    "early_payments":[
    {"on": moment("2015-04-15"),
     "amount": money("5000", "HKD")}
    ],
    "credit_draws":[
    {"on": moment("2015-09-15"),
     "amount": money(15000, "HKD")}
    ]})
calculator = new LoanCalculator.LoanCalculator();
calculator.calculate(my_term_sheet);
console.log()
console.log("Assume revenue hit on 2015-06-15 and 2015-09-15 and credit draws.  Skip payment on 2015-09-01");
my_term_sheet = new TermSheet.TermSheet();
my_term_sheet.set_events({
    "revenues": [
    {"on" : moment("2015-06-15"),
     "amount" : money("800000", "HKD")},
    {"on" : moment("2015-09-15"),
     "amount" : money("800000", "HKD")}
    ],
    "early_payments":[
    {"on": moment("2015-04-15"),
     "amount": money(5000, "HKD")}
    ],
    "credit_draws":[
    {"on": moment('2015-08-15'),
     "amount": money(15000, "HKD")}
    ],
    "skip_payments": [ moment('2015-09-01')]});
calculator = new LoanCalculator.LoanCalculator();
calculator.calculate(my_term_sheet);
